fabric-model-loading-api-v1/src/client/java/net/fabricmc/fabric/api/client/model/loading/v1/ModelModifier.java

```diff
@@ -0,0 +1,220 @@
+	 * e.g. for connected textures or other similar visual effects that should be the final processing step.
+	 */
+	public static final Identifier WRAP_LAST_PHASE = new Identifier("fabric", "wrap_last");
+
+	@FunctionalInterface
+	public interface OnLoad {
+		/**
+		 * This handler is invoked to allow modifying the unbaked model instance that is stored when it is first loaded.
+		 *
+		 * @param model the current unbaked model instance
+		 * @param context context with additional information about the model/loader
+		 * @return the model that should be used in this scenario. If no changes are needed, just return {@code model} as-is.
+		 * @see ModelLoadingPlugin.Context#modifyModelOnLoad
+		 */
+		UnbakedModel modifyModelOnLoad(UnbakedModel model, Context context);
+
+		/**
+		 * Context for an on load model modification event.
+		 */
+		@ApiStatus.NonExtendable
+		interface Context {
+			/**
+			 * The identifier of this model (may be a {@link ModelIdentifier}).
+			 *
+			 * <p>For item models, only the {@link ModelIdentifier} with the {@code inventory} variant is passed,
+			 * and not the corresponding plain identifier.
+			 */
+			Identifier id();
+
+			/**
+			 * Loads a model using an {@link Identifier} or {@link ModelIdentifier}, or gets it if it was already loaded.
+			 *
+			 * <p>Please note that the game engine keeps track of circular model loading calls on its own.
+			 *
+			 * @param id the model identifier
+			 * @return the unbaked model, or a missing model if it is not present
+			 */
+			UnbakedModel getOrLoadModel(Identifier id);
+
+			/**
+			 * The current model loader instance (changes when resource packs reload).
+			 *
+			 * <p>Do <b>not</b> call {@link ModelLoader#getOrLoadModel} as it doesn't supported nested model resolution,
+			 * use {@link #getOrLoadModel} from the context instead.
+			 *
+			 * @apiNote Calling {@link ModelLoader#getOrLoadModel} is fine for {@link BeforeBake} and {@link AfterBake},
+			 * it should only be avoided in {@link OnLoad} listeners.
+			 */
+			ModelLoader loader();
+		}
+	}
+
+	@FunctionalInterface
+	public interface BeforeBake {
+		/**
+		 * This handler is invoked to allow modifying the unbaked model instance that is baked.
+		 *
+		 * @param model the current unbaked model instance
+		 * @param context context with additional information about the model/loader
+		 * @return the model that should be used in this scenario. If no changes are needed, just return {@code model} as-is.
+		 * @see ModelLoadingPlugin.Context#modifyModelBeforeBake
+		 */
+		UnbakedModel modifyModelBeforeBake(UnbakedModel model, Context context);
+
+		/**
+		 * Context for a before bake model modification event.
+		 */
+		@ApiStatus.NonExtendable
+		interface Context {
+			/**
+			 * The identifier of this model (may be a {@link ModelIdentifier}).
+			 */
+			Identifier id();
+
+			/**
+			 * Function that can be used to retrieve sprites.
+			 */
+			Function<SpriteIdentifier, Sprite> textureGetter();
```

Comment:
Should this be `Sprite getTexture(SpriteIdentifier)` instead? Exposing functional interfaces as method return values is a bit strange.

Reply:
This mimics what `UnbakedModel` receives - as discussed on Discord it's fine either way.